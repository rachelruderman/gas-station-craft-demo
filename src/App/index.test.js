import React from 'react';
import { render } from '@testing-library/react';
import { App } from './index';

it('should take a snapshot', () => {
  const { asFragment } = render(<App />)
  expect(asFragment(<App />)).toMatchSnapshot()
});

test('renders Clearbit attribution', () => {
  const { getByText } = render(<App />);
  const element = getByText(/Logos provided by Clearbit/i);
  expect(element).toBeInTheDocument();
});

test('renders Fuel Up copyright', () => {
  const { getByText } = render(<App />);
  // todo: make dynamic w/ current year
  const element = getByText(/Fuel Up Â© 2020/i);
  expect(element).toBeInTheDocument();
});

