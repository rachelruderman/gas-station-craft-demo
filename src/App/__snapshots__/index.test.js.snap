// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    id="app"
  >
    <header>
      <h1>
        Running on empty?
      </h1>
      <h3>
        Take a break, grab a coffee and fuel up!
      </h3>
    </header>
    <div
      id="body"
    >
      <button
        class="initial"
        data-testid="cta-button"
        id="cta"
        type="button"
      >
        Find local gas stations
      </button>
    </div>
    <footer>
      <div>
        Fuel Up Â© 2020
      </div>
      <a
        href="https://clearbit.com"
      >
        Logos provided by Clearbit
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
